src/server/service/diagnostics/log.rs

```diff
@@ -233,9 +248,15 @@ fn parse_level(input: &str) -> IResult<&str, &str> {
 /// Parses the single log line and retrieve the log meta and log body.
 fn parse(input: &str) -> IResult<&str, (i64, LogLevel)> {
     let (content, (time, level, _)) = tuple((parse_time, parse_level, space1))(input)?;
-    let timestamp = match DateTime::parse_from_str(time, "%Y/%m/%d %H:%M:%S%.3f %z") {
-        Ok(t) => t.timestamp_millis(),
-        Err(_) => INVALID_TIMESTAMP,
+    let timestamp;
+    // return Err if parse time failed
+    match DateTime::parse_from_str(time, "%Y/%m/%d %H:%M:%S%.3f %z") {
+        Ok(t) => {
+            timestamp = t.timestamp_millis();
+        }
+        Err(_) => {
+            return Err(Err::Incomplete(Needed::Unknown));
```

Comment:
I think we should return an error directly instead of INVALID_TIMESTAMP value to keep the same logic as sysutil ( https://github.com/pingcap/sysutil/blob/master/search_log.go#L224 ). But  here I didn't find a very proper value to return, actually we don't care about its value, so I just choose the simplest one.

Reply:
I think return Err is ok. We can change the function signature to `fn parse(input: &str) -> Result<(&str, (i64, LogLevel)), Error>` then map errors to `Error::ParseError(err.to_string())`.

Reply:
Miss comment?

Reply:
Ok, let me have a try.

Reply:
Done!